extensions/ql-vscode/src/quick-query.ts

```diff
@@ -109,24 +106,22 @@ export async function displayQuickQuery(
       libraryPathDependencies: [qlpack]
     };
 
+    // Only recreate the query file if it doesn't already exist.
+    // Always recreate the qlpack file because the file will change when database languages change
```

Comment:
If the database language changes, then the existing query file may not be useful (especially if it contains `import <language>`). Can we detect whether the language has changed, and replace both query and qlpack files only in that case?

Reply:
I was hoping to avoid the complexity, but maybe because of the import statement, we do want to overwrite the old file.

Reply:
Simpler: always overwrite both query and qlpack files whenever you switch database. No need to check the language.

This is suboptimal if you're writing a quick query for the same language and want to run it on several different DBs. But it's an improvement on what we currently have. If the user needs the old query they can still get it from query history.

Reply:
The complexity is more that the user may change databases and keep the quick-query file open. When that happens, the contents will be wrong. The quick-query contents are only recreated when the command is called.

I guess this incorrect behaviour is currently in the extension. In order to fix this, I need to add a listener for database changes. We already have these listeners, so it's not too bad.

Reply:
I think that's ok? If they have the query open we ignore it. If they ask for a fresh Quick Query we give them a fresh one if it's a new DB, or the existing one if the DB hasn't changed. Seems like a strict win without having to add more listeners, though if you think that's worthwhile I don't mind.

Reply:
No, that's getting messy. I'll change things so that the quick query is only updated when the database changes (if necessary).